### ISSUE TITLE
[Bug]: update/delete table 1900w+ rows reported error ExpectedEOB

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#10107]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):7e6102854968449264be84501f18ee636dce1606
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

update table 1984w rows ,reported ExpectedEOB
![image](https://github.com/matrixorigin/matrixone/assets/113406637/a8fcce93-0a0b-4829-ba3b-6288bc52e965)

### Steps to Reproduce

```Markdown
on 128 server
create table table_100_columns_4000w( clo1 tinyint, clo2 smallint, clo3 int, clo4 bigint, clo5 tinyint unsigned, clo6 smallint unsigned, clo7 int unsigned, clo8 bigint unsigned, col9 float, col10 double, col11 varchar(255), col12 Date, col13 DateTime, col14 timestamp, col15 bool, col16 decimal(5,2), col17 text, col18 varchar(255), col19 varchar(255), col20 varchar(255), col21 varchar(255), col22 varchar(255), col23 varchar(255), col24 varchar(255), col25 varchar(255), col26 varchar(255), col27 varchar(255), col28 varchar(255), col29 varchar(255), col30 varchar(255), col31 varchar(255), col32 varchar(255), col33 varchar(255), col34 varchar(255), col35 varchar(255), col36 varchar(255), col37 varchar(255), col38
varchar(255), col39 varchar(255), col40 varchar(255), col41 varchar(255), col42 varchar(255), col43 varchar(255), col44 varchar(255), col45 varchar(255), col46 varchar(255), col47 varchar(255), col48 varchar(255), col49 varchar(255), col50 varchar(255), col51 varchar(255), col52 varchar(255), col53 varchar(255), col54 varchar(255), col55 varchar(255), col56 varchar(255), col57 varchar(255), col58 varchar(255), col59 varchar(255), col60 varchar(255), col61 varchar(255), col62 varchar(255), col63 varchar(255), col64 varchar(255), col65 varchar(255), col66 varchar(255), col67 varchar(255), col68 varchar(255), col69 varchar(255), col70 varchar(255), col71 varchar(255), col72 varchar(255), col73 varchar(255), col74 varchar(255), col75 varchar(255), col76 varchar(255), col77 varchar(255), col78 varchar(255), col79 varchar(255), col80 varchar(255), col81 varchar(255), col82 varchar(255), col83 varchar(255), col84 varchar(255), col85 varchar(255), col86 varchar(255), col87 varchar(255), col88 varchar(255), col89 varchar(255), col90 varchar(255), col91 varchar(255), col92 varchar(255), col93 varchar(255), col94 varchar(255), col95 varchar(255), col96 varchar(255), col97 varchar(255), col98 varchar(255), col99 varchar(255), col100 varchar(255) )partition by hash(clo3)partitions 4;

load data infile '/data/mo-load-data/data/100_columns/40000000_100_columns_load_data.csv' into table table_100_columns_4000w  FIELDS TERMINATED BY ',' LINES TERMINATED BY '\n' parallel 'true';
or s3 load:
load data url s3option {'bucket'='mo-load-data', 'filepath'='100_columns/40000000_100_columns_load_data.csv'} into table test.table_100_columns_4000w fields terminated by ',' lines terminated by '\n' parallel 'true';

update table_100_columns_4000w set clo3=20 where clo1>0;
```

**Comment by heni02** [comment]
same as delete:
![image](https://github.com/matrixorigin/matrixone/assets/113406637/d53f53dd-72c2-48f6-b523-a68fb14ca81d)

**Comment by heni02** [comment]
mo log：
[log (1).txt](https://github.com/matrixorigin/matrixone/files/11769509/log.1.txt)

**Comment by heni02** [comment]
@w-zr plz confirm issue

**Comment by JackTan25** [comment]
delete from table_100_columns_4000w doesn't pass through truncate? strange

**Comment by w-zr** [comment]
![image](https://github.com/matrixorigin/matrixone/assets/28624654/31be4a80-dede-45ec-ba84-8ddccf032700)
It's a commit error.

**Comment by reusee** [comment]
Looks like TAE commit failed, re-assigning to @XuPeng-SH

**Comment by XuPeng-SH** [comment]
will investigate it later

**Comment by XuPeng-SH** [comment]
@cnutshell please help to investigate this issue

**Comment by cnutshell** [comment]
Working on two other S-1 issue now.

**Comment by cnutshell** [comment]
The reported error is constructed via [moerr.GetOkExpectedEOB](https://github.com/matrixorigin/matrixone/blob/7e6102854968449264be84501f18ee636dce1606/pkg/common/moerr/error.go#L500).
And the error comes from [(*TableEntry).GetSegmentByID](https://github.com/matrixorigin/matrixone/blob/7e6102854968449264be84501f18ee636dce1606/pkg/vm/engine/tae/catalog/table.go#L170), detailed as follow:
<img width="930" alt="image" src="https://github.com/matrixorigin/matrixone/assets/20291742/9437fb0d-a62f-4c97-9f2e-b22fe258dea7">

**Comment by cnutshell** [comment]
On error `ExpectedEOB`, the transaction would rollback.
<img width="659" alt="image" src="https://github.com/matrixorigin/matrixone/assets/20291742/07a0c938-6b06-4693-9bb8-7db8aeb1136a">

**Comment by cnutshell** [comment]
Working on two other S-1 issues

**Comment by cnutshell** [comment]
I'll take this issue next week

**Comment by cnutshell** [comment]
Because 128 isn't convenient for debug, I've ask @heni02 for `40000000_100_columns_load_data.csv` in order to debug it in my local environment.

**Comment by cnutshell** [comment]
Test with commit `218775cedb02c211da040` locally, the error of `ExpectedEOB` still exists.
<img width="1439" alt="image-20230703220556618" src="https://github.com/matrixorigin/matrixone/assets/20291742/16f310b2-bd2f-48b6-9356-8de2201933ff">

**Comment by cnutshell** [comment]
The process was:
- `2023/07/05 16:26:01.947709`, segment A inserted by `catalog.(*TableEntry).AddEntryLocked`
- `2023/07/05 16:27:07.159725`, segment A wasn't found in `catalog.(*TableEntry).GetSegmentByID`

There wasn't delete event for segment A. I'm checking related code for more information.

**Comment by cnutshell** [comment]
@XuPeng-SH transfer this issue

**Comment by sukki37** [comment]
@XuPeng-SH please re-assign this issue, thank you.

**Comment by triump2020** [comment]
on vacation

**Comment by triump2020** [comment]
Pls ref to https://github.com/matrixorigin/matrixone/issues/10364

**Comment by triump2020** [comment]
blocked

**Comment by triump2020** [comment]
Blocked by other issue

**Comment by triump2020** [comment]
Blocked

**Comment by triump2020** [comment]
Blocked by other S-1

**Comment by triump2020** [comment]
blocked

**Comment by XuPeng-SH** [comment]
@triump2020 has another s-1 issue and assign it to @jiangxinmeng1

**Comment by jiangxinmeng1** [comment]
提交的时候会超时，还在定位超时的原因

**Comment by jiangxinmeng1** [comment]
fixed by #11294

**Comment by heni02** [comment]
confirm,closed
commit id:a977968783dc3fa1f5f64f2745026ebc45bcffd6
![image](https://github.com/matrixorigin/matrixone/assets/113406637/14a92a47-1c8e-4a5a-9e9f-30de1ba375c5)
